File: cpp/leetcode/058.cpp
class Solution {
  public:
    int lengthOfLastWord(string s) {
        int res = 0, idx = s.size() - 1;
        for (; idx >= 0 && s[idx] == ' '; --idx);
        for (; idx >= 0 && s[idx] != ' '; ++res, --idx);
        return res;
    }
};

File: python/leetcode/058.py
class Solution(object):
    def lengthOfLastWord(self, s):
        if not s:
            return 0
        i, res = len(s) - 1, 0
        while i >= 0 and s[i] == ' ':
            i -= 1
        while i >= 0 and s[i] != ' ':
            i -= 1
            res += 1
        return res
